### PR TITLE
ensure user metadata is created before link documents are created

### PR DESCRIPTION
**Pull request by shogunpurple**
## Description
Ongoing fixes branch:

Addresses: 
- https://github.com/Budibase/budibase/issues/2020
- https://github.com/Budibase/budibase/issues/1922
- https://github.com/Budibase/budibase/issues/2139
- https://github.com/Budibase/budibase/issues/2108

## Screenshots
_If a UI facing feature, some screenshots of the new functionality._

### CODE REVIEW

File: packages/bbui/src/Form/DatePicker.svelte [hunk 1]
```diff
@@ -13,8 +13,9 @@
 
   const dispatch = createEventDispatcher()
   const onChange = e => {
-    value = e.detail
-    dispatch("change", e.detail)
+    const isoString = e.detail.toISOString()
```
Comment (aptkingston) [inline packages/bbui/src/Form/DatePicker.svelte:16]: Was this the cause of the bad date sorting issue - that it was storing actual JS objects sometimes and strings other times? I know Flatpickr emits JS date objects (and I hate Flatpickr) but there could be consequences of changing this somewhere that might have been written to expect the date objects, so it just might need checked well.

Comment (shogunpurple) [reply packages/bbui/src/Form/DatePicker.svelte:16]: Yep, there was a mismatch between the way dates are stored in the system, between handlebars representation (which is ISO) and the way flatpickr stores it, which is just `new Date().toString()` from the looks of it. Updating this still seems to work with flatpickr and store the right format in the DB.

**Comment by kevmodrome** [comment]
Can this be merged? Would love to get #1922 fixed so I can make the tutorials look a bit better 👍

**Review by aptkingston — APPROVED** [review]

File: packages/bbui/src/Form/DatePicker.svelte (hunk 1 above)
Comment (aptkingston) [reply packages/bbui/src/Form/DatePicker.svelte:16]: Cool, all good then.

**Pull request merged**